lib/jxl/modular/transform/enc_palette.cc

```diff
@@ -298,19 +373,21 @@ Status FwdPalette(Image &input, uint32_t begin_c, uint32_t end_c,
           }
           if (found) break;
         }
-        if (index < nb_deltas) {
+        if (index < static_cast<int>(nb_deltas)) {
           delta_used = true;
         }
       } else {
         for (size_t c = 0; c < nb; c++) {
-          color_with_error[c] = p_in[c][x] + error_row[0][c][x + 2];
+          color_with_error[c] = p_in[c][x] + palette_iteration_data.final_run *
```

Comment:
Probably clearer (and faster) to write this as a ternary rather than as multiplication by a bool...

Reply:
I find it clearer like this, but if you provide any evidence about the better speed, I am happy to change it.